app/assets/src/components/visualizations/heatmap/Heatmap.js

```diff
@@ -511,6 +517,12 @@ export default class Heatmap {
 
     // Set up scrolling behavior.
     this.g.on("wheel.zoom", this.scroll.bind(this));
+
+    // If there's a specific row we want to focus on, auto-scroll the heatmap
+    // so the row is centered on screen.
+    if (this.focus) {
+      this.scrollToFocus();
```

Comment:
Could this be called from the SamplesHeatmapViz instead?
This feature seems very specific, and i would prefer to just enable the mechanism to scroll to a particular row (by index or by name).
That way heatmap would not have to worry about triggering/disabling this.
SamplesHeatmapViz would be responsible for keeping track of the focus row, choosing which one to scroll to, and asking that change from the heatmap.

Reply:
That makes a lot of sense, thanks for the suggestion! Just made the change.